### PR TITLE
Commander Core: Add fixed speed support

### PR DESCRIPTION
**Pull request by ParkerMc**
Adds pump and fan detection to initialize.
Adds support for setting a fixed percentage as the speed to the pump and each fan.

Related: #218 

---

Checklist:

- [X] Adhere to the [development process]
- [X] Conform to the [style guide]
- [X] Verify that the changes work as expected on real hardware
- [x] Add automated tests cases
- [x] Verify that all (other) automated tests (still) pass
- [x] Update the README and other applicable documentation pages
- [x] Update the `liquidctl.8` Linux/Unix/Mac OS man page
- [x] Update or add applicable `docs/*guide.md` device guides
- [x] Submit relevant data, scripts or dissectors to https://github.com/liquidctl/collected-device-data

New CLI flag?

- [ ] Adjust the completion scripts in `extra/completions/`

New device?

- [ ] Regenerate `extra/linux/71-liquidctl.rules` (instructions in the file header)
- [ ] Add entry to the README's supported device list with applicable notes (at least `en`)

New driver?

- [x] Document the protocol in `docs/developer/protocol/`

[development process]: https://github.com/liquidctl/liquidctl/blob/main/docs/developer/process.md
[style guide]: https://github.com/liquidctl/liquidctl/blob/main/docs/developer/style-guide.md

### CODE REVIEW

**Comment by ParkerMc** [comment]
Still working on documentation, tests, and updating collected device data, but everything else is ready for feedback.

File: liquidctl/driver/commander_core.py [hunk 1]
```diff
@@ -58,15 +67,21 @@ def initialize(self, **kwargs):
...
+                connected = u16le_from(led_data, offset=i * 4) == 2
+                num_leds = u16le_from(led_data, offset=i * 4 + 2)
                 label = 'AIO LED count' if i == 0 else f'RGB port {i} LED count'
                 status += [(label, num_leds if connected else None, '')]
 
+            # Get what fans are connected
+            res = self._read_data(_MODE_CONNECTED_SPEEDS, _DATA_TYPE_CONNECTED_SPEEDS)
+            num_devices = res[0]
+            for i in range(0, num_devices):
+                label = 'AIO Pump' if i == 0 else f'Fan port {i}'
```
Comment (jonasmalacofilho) [inline liquidctl/driver/commander_core.py:82]: Companion to previous nitpick: `AIO pump`

**Review by jonasmalacofilho — COMMENTED** [review]
The implementation looks great.  I especially like how you were able to only change the mode of the specified channel, _without storing anything on the liquidctl side._

Just one comment on the documentation, and a nitpick on the casing of a status attribute name.

File: docs/corsair-commander-core-guide.md [hunk 2]
```diff
@@ -10,14 +10,21 @@ The device should be initialized every time it is powered on.
...
-├── Firmware version            1.6.135  
+├── Firmware version            2.6.201  
 ├── AIO LED count                    29  
 ├── RGB port 1 LED count              8  
 ├── RGB port 2 LED count              8  
 ├── RGB port 3 LED count            N/A  
 ├── RGB port 4 LED count            N/A  
 ├── RGB port 5 LED count            N/A  
 ├── RGB port 6 LED count            N/A  
+├── AIO Pump                        Yes
```
[the remark this answers is not in the document]
Comment (ParkerMc) [reply docs/corsair-commander-core-guide.md:21]: Should AIO pump become just AIO?

File: docs/corsair-commander-core-guide.md [hunk 3]
```diff
@@ -39,3 +46,19 @@ Corsair Commander Core (experimental)
...
+# liquidctl set fan1 speed 70
+                ^^^^       ^^
+               channel    duty
+```
+
+Valid channel values are `pump`, `fanN`, where 1 <= N <= 6 is the fan number, and
+`fans`, to simultaneously configure all fans.
+
+Note: There is a hardware minimum speed when setting a fixed duty cycle. Anything below that point just runs at the minimum.
+This means that 0 is not stopped and a lot of the lower speeds mean the same speed.
```
[the remark this answers is not in the document]
Comment (ParkerMc) [reply docs/corsair-commander-core-guide.md:64]: So iCUE, limits between 0% and 20%. I can set it to 0% or anything 20% and higher. I have two different types of fans now ML and LL, when set to 0% the ML's will stop and the LL's will not. So I'll have to test the differences more thoroughly.
As for the pump, iCUE limits it to 3 preset settings. `Quiet`, `Balanced`, `Extreme`, and `Variable Speed`. I will figure out what exactly those are and test lower ones to see how exactly the pump responds.

[the remark this answers is not in the document]
Comment (jonasmalacofilho) [reply docs/corsair-commander-core-guide.md:64]: Oh, I see.  In that case, let's leave them unrestrained for now.

File: docs/corsair-commander-core-guide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonasmalacofilho) [reply docs/corsair-commander-core-guide.md:21]: I think that makes sense.  Or maybe even `AIO port`.

Also, since these are all booleans, maybe `AIO port connected` or `AIO port in use`?  (And similarly for the fan ports).

File: liquidctl/driver/commander_core.py [hunk 4]
```diff
@@ -99,7 +114,27 @@ def set_speed_profile(self, channel, profile, **kwargs):
         raise NotSupportedByDriver
 
     def set_fixed_speed(self, channel, duty, **kwargs):
```
Comment (ParkerMc) [inline liquidctl/driver/commander_core.py:116]: Any thoughts on displaying the fan/pump mode and target fixed speed (if set) that are set here in the status message? Maybe potentially under `--verbose`? If so is there a device that does this already or format ideas?
This can be another PR, I just had the idea.

File: docs/corsair-commander-core-guide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ParkerMc) [reply docs/corsair-commander-core-guide.md]: I rewrote that note so it makes more sense.

**Comment by ParkerMc** [comment]
> I especially like how you were able to only change the mode of the specified channel, _without storing anything on the liquidctl side._

That was actually a goal I had because I really don't want it to change speed and mess up the fan curve when I switch between windows and linux unless I specifically tell it to. Protocol made it decently easy, just have to add the read and modify the data.

File: liquidctl/driver/commander_core.py (hunk 4 above)
Comment (jonasmalacofilho) [reply liquidctl/driver/commander_core.py:116]: That would be nice.

For the pump/fan mode, take a look at the Commander Pro and Smart Device (V1): those specific devices only report PWM or DC, but you could extend that idea for whether their running in fixed PWM or on-board profile modes.  IIRC, they use enums with a suitable `__str__()` implementation for the pretty/human output.

For the target duty cycle (if set), take a look at the Corsair Platinum coolers.

**Review by jonasmalacofilho — APPROVED** [review]

**Comment by jonasmalacofilho** [comment]
Thanks!
